src/promptflow/promptflow/_cli/_pf/_upgrade.py

```diff
@@ -0,0 +1,160 @@
+import os
+from promptflow._cli._params import (
+    add_param_yes,
+    base_params,
+)
+from promptflow._cli._utils import activate_action, get_cli_sdk_logger
+from promptflow._utils.utils import prompt_y_n
+from promptflow.exceptions import UserErrorException
+
+logger = get_cli_sdk_logger()
+
+UPGRADE_MSG = 'Not able to upgrade automatically'
+
+
+def add_upgrade_parser(subparsers):
+    """Add upgrade parser to the pf subparsers."""
+    epilog = """
+    Examples:
+
+    # Upgrade prompt flow without prompt and run non-interactively:
+    pf upgrade --yes
+    """  # noqa: E501
+    add_params = [
+                     add_param_yes,
+                 ] + base_params
+    activate_action(
+        name="upgrade",
+        description="Upgrade prompt flow CLI.",
+        epilog=epilog,
+        add_params=add_params,
+        subparsers=subparsers,
+        help_message="pf upgrade",
+        action_param_name="action",
+    )
+
+
+def upgrade_version(args):
+    import platform
+    import sys
+    import subprocess
+
+    from promptflow._constants import _ENV_PF_INSTALLER, CLI_PACKAGE_NAME
+    from promptflow._version import VERSION as local_version
+    from packaging.version import parse
+
+    from promptflow._utils.version_hint_utils import get_latest_version_from_pypi
+    latest_version = get_latest_version_from_pypi(CLI_PACKAGE_NAME)
+    if not latest_version:
+        logger.warning("Failed to get the latest prompt flow version.")
+        return
+    elif parse(latest_version) <= parse(local_version):
+        logger.warning("You already have the latest prompt flow version: %s", local_version)
+        return
+
+    yes = args.yes
+    exit_code = 0
+    installer = os.getenv(_ENV_PF_INSTALLER) or ''
+    installer = installer.upper()
+    print(f"installer: {installer}")
+    latest_version_msg = 'It will be updated to {}.'.format(latest_version) if yes \
+        else 'Latest version available is {}.'.format(latest_version)
+    logger.warning("Your current Azure CLI version is %s. %s", local_version, latest_version_msg)
+    if not yes:
+        logger.warning("Please check the release notes first")
+        if not sys.stdin.isatty():
+            logger.debug('No tty available.')
+            raise UserErrorException("No tty available. Please run command with --yes.")
+        confirmation = prompt_y_n("Do you want to continue?", default='y')
+
+        if not confirmation:
+            logger.debug("Upgrade stopped by user")
+            return
+
+    if installer == 'MSI':
+        _upgrade_on_windows()
```

Comment:
can we pass in the yes arg to msi install comamnd?

Reply:
why we need pass yes to msi?